libs/java/service/src/main/java/ca/bc/gov/tno/services/kafka/handlers/ConsumerPauseHandler.java

```diff
@@ -0,0 +1,46 @@
+package ca.bc.gov.tno.services.kafka.handlers;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.ApplicationListener;
+import org.springframework.scheduling.annotation.Async;
+import org.springframework.stereotype.Service;
+
+import ca.bc.gov.tno.services.ServiceState;
+import ca.bc.gov.tno.services.ServiceStatus;
+import ca.bc.gov.tno.services.kafka.events.ConsumerPauseEvent;
+
+/**
+ * PauseHandler class, provides a listener for the ConsumerPauseEvent. It will
+ * change the service state to pause.
+ */
+@Async
+@Service
```

Comment:
I'd love to know how to make optional beans in java.  Presently this would add a listener in an package that includes this library...  Spring is crap...